File: state_machine/lambdas/archive/send_sns.py
import json
import boto3

import logging
# Initialize logger and set log level
logger = logging.getLogger()
logger.setLevel(logging.INFO)

print('Loading send_sns function...')
sns = boto3.client('sns')


def lambda_handler(event, context):
    # Log the received event
    #print("Received event: " + json.dumps(event, indent=2))
    email_message = "Hey man....  Your batch schedule: " + event['scheduleName'] + " [started: " + event['startDateTime'] + "] just finished.  Status: " + event['scheduleStatus']
    sms_message = "Job Complete. Status = " + event['scheduleStatus']
    sns_arn = event['scheduleSnsTopic']
    subject = 'Batch schedule ' + event['scheduleStatus']

    # Send message
    response = sns.publish(
        TargetArn=sns_arn,
        Message=json.dumps({'default': json.dumps(email_message),
                            'sms': json.dumps(sms_message),
                            'email': json.dumps(email_message)}),
        Subject=subject,
        MessageStructure='json'
    )

    logger.info(response)
    print('SNS function complete')
    print("Returned event: " + json.dumps(event, indent=2))
    return event

File: state_machine/lambdas/archive/send_sns.js
console.log('Loading function');
const aws = require('aws-sdk');

exports.lambda_handler = (event, context, callback) => {
    const sns = new aws.SNS();
    console.log(`Sending message to SNS topic ${event.ScheduleSnsTopic}`);
    const messageInfo = {
        Message: event.ScheduleMessage,
        TopicArn: event.ScheduleSnsTopic       
    };
    sns.publish(messageInfo, (err, data) => {
        if (err) {
            console.error(err.message);
            callback(err.message, event);
            return;
        }
        console.log(data);
        callback(err, event);
    });
};